File: python-venv/simple_string_expansion/kata_function/simple_string_expansion.py
import re

def string_expansion(string):

    if len(re.findall("[0-9]", string)) == 0:
        return string

    output = ''
    first = False

    for i in range(len(string)):
        if len(re.findall("[0-9]", string[i])) > 0:
            first = True
            for j in range(i + 1, len(string), 1):
                if len(re.findall("[0-9]", string[j])) > 0:
                    break
                output += string[j] * int(string[i])

        if first == False:
            output += string[i]

    return output

#https://www.codewars.com/kata/5ae326342f8cbc72220000d2/train/python

File: simple-string-expansion/simpleStringExpansion.js
const stringExpansion = (s) => {

    let output = '';

    if(s.search(/[0-9]/g) === -1) return s;

    for(let k = 0; k < s.length; k++) {
        if(/[0-9]/g.test(Number(s[k])) === false) {
            output += s[k];
        } else {
            break;
        };
    };

    for(let i = 0; i < s.length; i++) {
        if(/[0-9]/g.test(Number(s[i])) === true) {
            for(let j = i + 1; j < s.length; j++) {
                if(s[j] === undefined) break;
                if(/[0-9]/g.test(Number(s[j])) === true) break;
                let append = s[j].repeat(s[i]);
                output += append;
            };
        };
    };
    
    return output;

};

module.exports = { stringExpansion };

//https://www.codewars.com/kata/5ae326342f8cbc72220000d2/javascript